src/main/java/htsjdk/samtools/cram/compression/rans/ransnx16/RANSNx16Decode.java

```diff
@@ -0,0 +1,488 @@
+        }
+        for (int i = 0; i< numRLESymbols; i++) {
+            rleSymbols[uncompressedRLEMetaData.get() & 0xFF] = 1;
+        }
+        return uncompressedRLEMetaData;
+    }
+
+    private ByteBuffer decodeRLE(
+            ByteBuffer inBuffer,
+            final int[] rleSymbols,
+            final ByteBuffer uncompressedRLEMetaData,
+            final int uncompressedRLEOutputLength) {
+        ByteBuffer rleOutBuffer = ByteBuffer.allocate(uncompressedRLEOutputLength);
+        int j = 0;
+        for(int i = 0; j< uncompressedRLEOutputLength; i++){
+            byte sym = inBuffer.get(i);
+            if (rleSymbols[sym & 0xFF]!=0){
+                int run = Utils.readUint7(uncompressedRLEMetaData);
+                for (int r=0; r<= run; r++){
+                    rleOutBuffer.put(j++, sym);
+                }
+            }else {
+                rleOutBuffer.put(j++, sym);
+            }
+        }
+        inBuffer = rleOutBuffer;
+        return inBuffer;
+    }
+
+    private ByteBuffer decodePack(
+            ByteBuffer inBuffer,
+            final int[] packMappingTable,
+            final int numSymbols,
+            final int uncompressedPackOutputLength) {
+        ByteBuffer outBufferPack = ByteBuffer.allocate(uncompressedPackOutputLength);
+        int j = 0;
+
+        if (numSymbols <= 1) {
+            for (int i=0; i < uncompressedPackOutputLength; i++){
+                outBufferPack.put(i, (byte) packMappingTable[0]);
+            }
+        }
+
+        // 1 bit per value
+        else if (numSymbols <= 2) {
+            int v = 0;
+            for (int i=0; i < uncompressedPackOutputLength; i++){
+                if (i % 8 == 0){
+                    v = inBuffer.get(j++);
+                }
+                outBufferPack.put(i, (byte) packMappingTable[v & 1]);
+                v >>=1;
+            }
+        }
+
+        // 2 bits per value
+        else if (numSymbols <= 4){
+            int v = 0;
+            for(int i=0; i < uncompressedPackOutputLength; i++){
+                if (i % 4 == 0){
+                    v = inBuffer.get(j++);
+                }
+                outBufferPack.put(i, (byte) packMappingTable[v & 3]);
+                v >>=2;
+            }
+        }
+
+        // 4 bits per value
+        else if (numSymbols <= 16){
+            int v = 0;
+            for(int i=0; i < uncompressedPackOutputLength; i++){
+                if (i % 2 == 0){
+                    v = inBuffer.get(j++);
+                }
+                outBufferPack.put(i, (byte) packMappingTable[v & 15]);
+                v >>=4;
+            }
+        }
+        inBuffer = outBufferPack;
+        return inBuffer;
+    }
+
+    private ByteBuffer decodeStripe(final ByteBuffer inBuffer, final int outSize){
+        final int numInterleaveStreams = inBuffer.get() & 0xFF;
+
+        // retrieve lengths of compressed interleaved streams
+        final int[] compressedLengths = new int[numInterleaveStreams];
```

Comment:
Something seems wrong here - this array is never used (never read from).

Reply:
yes, I agree. Following the spec, we get the lengths of compressed streams but these lengths are not being used anywhere. 

![image](https://github.com/samtools/htsjdk/assets/89084962/cd6e5d1e-9db3-4fa0-b60c-6ca4712f650f)
![image](https://github.com/samtools/htsjdk/assets/89084962/112b52b6-773d-4f1b-a8c0-6e02dc647186)

Reply:
As discussed, I am reading the compressed lengths but not allocating them to variables